zomes/likes/code/src/hdk_membrane.rs

```diff
@@ -0,0 +1,10 @@
+struct MembraneObject {
+
+}
+
+trait MembraneObject {
+    pub fn get_source_chain() -> Vec<Pair>;
+    pub fn query_source_chain(entry_type: String) -> Vec<Pair>;
+    pub fn agent_key() -> String;
+    pub fn agent_id() -> String;
+}
```

Comment:
I can't imagine how that would work. Rust macros can't differentiate between Traits and Structs. There is no way to make this OOP-like interface

Reply:
I'm not sure what you mean.. this is just for context so reading this PR makes sense. This should actually go to HDK as @ddd-mtl noticed. I just wanted to show which functions we'd provide to the membrane check callback. (This is the type of the parameter passed to any membrane callback).

This is clearly incomplete and won't compile. The main intention of this PR is to show an example of membranes so we can speak and reason about it and gauge consequences for the p2p/network implementation.

(get the feeling I should delete this file because it rather confuses than helps understanding ;D)